### PR TITLE
[CELEBORN-1312] Move handleRequestPartitions out of sync block

### PR DESCRIPTION
**Pull request by waitinfuture**
### What changes were proposed in this pull request?
As title, `handleRequestPartitions` is quite heavy since it calls sync RPC.
It's unnecessary to put it in the sync block.
This fixes the same issue as https://github.com/apache/incubator-celeborn/pull/2207

### Why are the changes needed?
ditto

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
GA and manual test.

### CODE REVIEW

**Comment by waitinfuture** [comment]
@lyy-pineapple PTAL, cc @RexXiong @AngersZhuuuu

**Comment by CodingCat** [comment]
+1, I am exactly looking at this yesterday

File: client/src/main/scala/org/apache/celeborn/client/ChangePartitionManager.scala [hunk 1]
```diff
@@ -19,7 +19,7 @@ package org.apache.celeborn.client
 
 import java.util
 import java.util.{Set => JSet}
-import java.util.concurrent.{ConcurrentHashMap, ScheduledExecutorService, ScheduledFuture, TimeUnit}
+import java.util.concurrent.{ConcurrentHashMap, ExecutorService, ScheduledExecutorService, ScheduledFuture, TimeUnit}
```
Comment (mridulm) [inline client/src/main/scala/org/apache/celeborn/client/ChangePartitionManager.scala:22]: nit: Not required ?

Comment (SteNicholas) [reply client/src/main/scala/org/apache/celeborn/client/ChangePartitionManager.scala:22]: +1

**Review by CodingCat — APPROVED** [review]

**Comment by waitinfuture** [comment]
Thanks, merging  to main(v0.5.0)/branch-0.4(v0.4.1)

File: client/src/main/scala/org/apache/celeborn/client/ChangePartitionManager.scala (hunk 1 above)
Comment (waitinfuture) [reply client/src/main/scala/org/apache/celeborn/client/ChangePartitionManager.scala]: fixed, thanks

**Pull request closed without merging**